nix/keys.nix

```diff
@@ -6,13 +6,38 @@ let
   keyOptionsType = types.submodule {
     options.text = mkOption {
       example = "super secret stuff";
-      type = types.str;
+      default = null;
+      type = types.nullOr types.str;
       description = ''
-        The text the key should contain. So if the key name is
-        <replaceable>password</replaceable> and <literal>foobar</literal>
-        is set here, the contents of the file
+        When non-null, this designates the text that the key should
+        contain. So if the key name is <replaceable>password</replaceable>
+        and <literal>foobar</literal> is set here, the contents of the file
         <filename>/run/keys/<replaceable>password</replaceable></filename>
         will be <literal>foobar</literal>.
+
+        NOTE: Either <literal>text</literal> or <literal>keyFile</literal> have
+        to be set.
+      '';
+    };
+
+    options.keyFile = mkOption {
+      example = ./super-secret.key;
```

Comment:
Hmm, oddly it never had given me a problem.  What was the error message?

But ok, removed it.